circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/annotation/TableAnnotator.java

```diff
@@ -0,0 +1,21 @@
+package com.hotels.bdp.circustrain.core.annotation;
+
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.Table;
+
+import com.hotels.bdp.circustrain.api.CircusTrainException;
+
+public interface TableAnnotator {
```

Comment:
Not quite sure I understand why you need this and `HiveTableAnnotator`

Reply:
There are two main paths, destructive and upsert. Destructive deletes the table without creating an intermediary table object (so the params would need to be added to the table in Hive directly), but upsert does create an intermediary java object, so I thought for the upsert path it would be easier to add params the params before it is then created in Hive.

Looking at it again perhaps I'm wrong, in the upsert path once the replica table is created in `Replica`, the params can be added using the `HiveTableAnnotator`. This would be just as easy.

Does that make sense?

Reply:
Removed `TableAnnotator` now and just have the `HiveTableAnnotator`.